src/Service/BookingRuleApplied.php

```
@@ -187,24 +188,26 @@ public static function bookingConformsToRules( Booking $booking):void {
 	}
 
 	/**
-	 * Gets a string of all rule properties, so they can be displayed using CMB2
+	 * Retrieves a JSON string of all rule properties to display using CMB2.
 	 *
-	 * Will ignore errors, so that the settings page can still display the selected values even if they are invalid
-	 * @return string
+	 * If initialization or encoding fails, returns an empty string to ensure the page functions.
+	 *
+	 * @return string JSON-encoded rule properties, or an empty string on failure.
 	 */
 	public static function getRulesJSON(): string {
-		$ruleObjects = static::init( true );
-
-		if ( isset( $ruleObjects ) ) {
-			return wp_json_encode(
-				array_map(
-					function( $rule){
-						return get_object_vars($rule);
-					}, $ruleObjects )
-			);
-		}
-		else {
-			return "";
+		try {
+			$ruleObjects = static::init(true);
```

Hier können durch die gesetzte ignore errors flag gar keine Exceptions geworfen werden.

---

In static::init wird noch mal `parent::init()` aufgerufen, was auch eine BookingException wirft. Daher finde ich den try-catch gerechtfertigt. Mir ging es mit der Änderung auch nur um den Aufrufer Teil hier. Ansonsten müsste man die init Methode mal refactoren.

Ich habe es nochmal etwas leserlicher geschrieben.